sopel_modules/twitter/twitter.py

```diff
@@ -144,7 +144,7 @@ def format_time(bot, trigger, stamp):
         bot.db, bot.config, tz, trigger.nick, trigger.sender, parsed)
 
 
-@module.url(r'https?://(?:m(?:obile)?\.)?twitter\.com/(?P<user>[^/]+)(?:$|/status/(?P<status>\d+)).*')
+@module.url(r'https?://(?:m(?:obile)?\.)?twitter\.com/(?P<user>[^/?]+)(?:/?(?:\?.*)?$|/status/(?P<status>\d+))')
```

Comment:
On further analysis, am I missing some reason we can't just do this? The code looks up the status ID if that group captured something, and looks up the user otherwise.

```suggestion
@module.url(r'https?://(?:m(?:obile)?\.)?twitter\.com/(?P<user>\w+)(?:/status/(?P<status>\d+))?')
```

Results:

![image](https://user-images.githubusercontent.com/164140/155786166-47dae8a7-41a1-4ec5-91bf-f3b29d2cc246.png)

Twitter usernames only allow alphanumeric + `_` anyway, so `\w` is perfect. I _wish_ matching valid GitHub user/repo names was that easy!

Reply:
Seems simpler and sturdier. :+1: